### PR TITLE
Db enummification

### PR DESCRIPTION
**Pull request by bnaecker**

### CODE REVIEW

**Comment by bnaecker** [comment]
This does some small cleanup towards finishing #312. It adds the instance, disk, and saga state enums in the database schema, and adds conversions for them.

Note that the `disk_state` and `saga_node_event_type` enums are probably going to be harder. Those carry data in some of their variants, which is usually recorded in a separate column in their tables. It should be possible to use a struct with two fields, and then `#[diesel(embed)]` that into the parent struct. I've opted to defer that for now.

File: common/src/sql/dbinit.sql [hunk 1]
```diff
@@ -636,23 +636,22 @@ CREATE UNIQUE INDEX ON omicron.public.saga (
...
- * TODO See notes above about cockroachdb/cockroach#57411 /
- * cockroachdb/cockroach#58084.
- */
--- CREATE TYPE omicron.public.SagaNodeEventType AS ENUM (
---     'started',
---     'succeeded',
---     'failed'
---     'undo_started'
---     'undo_finished'
+ * TODO: This is a data-carrying enum, see note on disk_state.
```
Comment (smklein) [inline common/src/sql/dbinit.sql:640]: Nit: link to https://github.com/oxidecomputer/omicron/issues/312?

**Review by davepacheco — APPROVED** [review]

File: common/src/sql/dbinit.sql [hunk 2]
```diff
@@ -281,7 +277,13 @@ CREATE UNIQUE INDEX ON omicron.public.instance (
  */
 
 /*
- * TODO See the note on InstanceState above.
+ * TODO The Rust enum to which this type is converted
```
[the remark this answers is not in the document]
Comment (bnaecker) [reply common/src/sql/dbinit.sql:280]: Done in f73fae5

File: common/src/sql/dbinit.sql (hunk 1 above)
Comment (bnaecker) [reply common/src/sql/dbinit.sql:642]: Done in f73fae5

**Pull request merged**